CTFd/api/v1/notifications.py

```diff
@@ -68,9 +72,26 @@ def get(self, query_args):
         field = str(query_args.pop("field", None))
         filters = build_model_filters(model=Notifications, query=q, field=field)
 
-        notifications = (
-            Notifications.query.filter_by(**query_args).filter(*filters).all()
+        notifications = Notifications.query.filter_by(**query_args).filter(*filters)
+
+        # Filter on user/team id if present in the notification
+        notifications = notifications.filter(
+            or_(
+                Notifications.user_id.is_(None),
+                Notifications.user_id == get_current_user().id,
+            )
         )
+        if get_config("user_mode") == TEAMS_MODE:
+            notifications = notifications.filter(
+                or_(
+                    Notifications.team_id.is_(None),
+                    Notifications.team_id == get_current_team().id,
+                )
+            )
+
+        # Grab the full list
+        notifications = notifications.all()
+
```

Comment:
This kind of looks unfortunate. Can you add onto the existing filters? And how does the user distinguish between notifications to them vs globally?